### PR TITLE
Issue 331: Add partial matching options to setup-options

### PR DESCRIPTION
**Pull request by Bisaloo**
## Description

Follow up of #338 

This PR is opened as a draft because checks are expected to fail. There are instances of partial matching in abind and since this package is not hosted on a public git repo, I can't submit a PR.

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L68

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L105

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L118

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L149

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L195

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L204

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L212

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L221-L222

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L233

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L250

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L277

https://github.com/cran/abind/blob/7d7185af7c3b3c487be6dcd2a4a69224d596ff0a/R/abind.R#L283

I'll keep thinking a little bit about this and see if the options can be turn off for `test-summary.epinowcast.R` only

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [ ] I have read the [contribution guidelines](https://github.com/epinowcast/epinowcast/blob/main/CONTRIBUTING.md).
- [ ] I have tested my changes locally.
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [ ] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have updated the package development version by one increment in both `NEWS.md` and the `DESCRIPTION`.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Comment by Bisaloo** [comment]
Unfortunately, we still have the same issue that warnings are easily missed on CI.

We could:

- decide that it's fine because contributors are asked to run tests locally anyways
- turn testthat warnings to errors globally but this has wider downstream effects

**Comment by seabbs** [comment]
Amazing thanks @Bisaloo! 

Hmm `abind` not having a public facing repo is very much not ideal. 

> turn testthat warnings to errors globally but this has wider downstream effects

I sort of think this is the way forward as I would prefer checks to be as automated as possible. Aside from impacting other potential warnings that have been going under the radar (which is not a bad thing) what other downstream effects do you see from this?

**Comment by Bisaloo** [comment]
> Aside from impacting other potential warnings that have been going under the radar (which is not a bad thing)

That was my main worry. Some projects are fine with letting some warnings slip through. It doesn't seem to be the case here but just flagging that changing this setting slightly reduce flexibility.

For the record, I have confirmed upgrading warnings to errors does not interfere with `testthat::expect_warning()`

``` r
options(warn = 2)

f <- function() {
  warning("This is a warning")
}

f()
#> Error in f(): (converted from warning) This is a warning

testthat::expect_warning(f())
```

<sup>Created on 2023-10-25 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by seabbs** [comment]
Just circling back to this. Is this waiting on a decision about 

> a small interaction with testthat::expect_error() now passing on simple warnings:

?

**Comment by Bisaloo** [comment]
> Is this waiting on a decision about

It's mostly waiting on my confusion to dissipate :grimacing:. The latest commit converts testthat warning to errors without any side-effect and seems to be working locally but CI checks are still passing, which should not be the case.
I haven't had the time to investigate but it's still on my todo list.

**Comment by seabbs** [comment]
AH okay got you! Note that #360 means that some of the CI checks are pretty unreliable right now.

**Comment by Bisaloo** [comment]
@seabbs, this is now behaving as expected but tests are failing because `options()` and environment variables are modified by epinowcast calls, which probably should not be the case.

What do you think?

**Review by seabbs — COMMENTED** [review]
Aside from the CI issues due to changing environment variables and the lack of news items this looks good to go. 

@Bisaloo do you think we should deal with the failing tests here or in a separate PR?

**Comment by Bisaloo** [comment]
This is almost ready :crossed_fingers:

Could a macOS user verify they also get the extra cpp11 `options()` please? I'm trying to understand if this is a macOS issue or if it's specific to the CI setup.

**Comment by seabbs** [comment]
> Could a macOS user verify they also get the extra cpp11 options() please? I'm trying to understand if this is a macOS issue or if it's specific to the CI setup.

I'll do this on my intel mac but does anyone have a Mn mac to try this out on (maybe @jamesmbaazam?)

**Comment by Bisaloo** [comment]
In order to move forward with this issue, I propose we comment out this line for now with a link to a new GitHub issue describing the issue and calling for further investigation in the future:

https://github.com/Bisaloo/epinowcast/blob/a31eb7402537e6f9fd4d1b024789e7c60575a9f2/tests/testthat/helper-state.R#L17

How does this sound?

**Comment by seabbs** [comment]
yes agree sounds good

**Comment by seabbs** [comment]
I'm again a bit lost on this. @Bisaloo do you have a good overview of what needs to be done/what this PR is waiting on?

**Comment by Bisaloo** [comment]
I expected it to be simple but it keeps hitting all these random issues. Let me restart the checks to see the current status.

**Comment by seabbs** [comment]
> expected it to be simple but it keeps hitting all these random issues

I might adopt this as a tagline

**Comment by Bisaloo** [comment]
I'm also not entirely sure why checks are not triggered :thinking:

**Comment by seabbs** [comment]
its the stupid README bot causing problems :(

**Comment by Bisaloo** [comment]
But closing & reopening the PR usually triggers them :thinking:

**Comment by Bisaloo** [comment]
Is it possible that CI from fork has been disabled? Should I try and push a new branch to this repo?

**Comment by seabbs** [comment]
> Is it possible that CI from fork has been disabled? Should I try and push a new branch to this repo?

No I just checked this and its only disabled for new contributors who are new to GitHub. Maybe try and empty commit to give it a kick?

File: README.md [hunk 1]
```diff
@@ -194,6 +194,8 @@ list.files(
 
 </details>
 
+## Contributors
```
Comment (seabbs) [inline README.md:197]: Hmm not really clear why this is happening (this is from all contributors and not you). Can you remove the README updates as this should now only trigger as PR anyways on main (so we can handle issues like that there)

**Review by seabbs — APPROVED** [review]
This looks good to me. Thanks for this @Bisaloo and sorry its been a bit of a pain. Could you remove the automatic readme update and then I think this is good to go.

**Comment by Bisaloo** [comment]
I see two issues:

- https://github.com/epinowcast/epinowcast/blob/90f7a5a05892a9d176cc71422b796b1c54aa24d6/R/model-tools.R#L222-L224 doesn't catch all the potential warnings as the last dir in the tree may not exist but the parents may, hence warnings when we use `recursive  = TRUE`
- there is a strange (and unexpected?) nesting of temp folders (e.g., `'/tmp/Rtmp57LXYf/working_dir/RtmpZ9eqUZ/include_1/RtmpZ9eqUZ/.' already exists`). I suspect this is due to the fact there are new R workers started at some point and `tempdir()` is given a different value. The `recursive = TRUE` means that the old `tempdir()` path is re-created in the new `tempdir()`.

**Comment by Bisaloo** [comment]
The two remaining failing tests are unrelated to this PR as far as I can tell.

**Comment by jamesmbaazam** [comment]
I think this PR is related to this issue I raised https://github.com/epinowcast/epinowcast/issues/472.

**Review by seabbs — APPROVED** [review]
Phew! This looks good I think. really nice to have this in place but what a journey.

**Pull request merged**